src/tools/lib_fs_utils.rs

```diff
@@ -173,12 +175,17 @@ fn cp_into_lind(cage: &Cage, hostfilepath: &interface::RustPath, lindfilepath: &
 
     let lindfd = cage.open_syscall(lindtruepath.to_str().unwrap(), O_CREAT | O_TRUNC | O_WRONLY, S_IRWXA);
     assert!(lindfd >= 0);
-    let veclen = filecontents.len();
-    let fileslice = filecontents.as_slice();
-    let writtenlen = cage.write_syscall(lindfd, fileslice.as_ptr(), veclen);
 
+    let veclen = filecontents.len();
+    let mut writtenlen: usize = 0;
+    
+    //on Linux, write() (and similar system calls) will transfer at most 0x7ffff000 (2,147,479,552) bytes
+    //dividing filecontents into chunks of 0x7ffff000 (2,147,479,552) bytes and writing each chunk
+    for chunk in filecontents.chunks(LINUX_MAX_RW_COUNT) {
+        writtenlen += cage.write_syscall(lindfd, chunk.as_ptr(), chunk.len()) as usize;
```

Comment:
We could consider keeping `total_written_len` and `current_written_len` as two separate variables, for clarity while reading the code.

(We should only do this if the compiler optimizes it so that it does not impact performance)